File: src/test/java/no/nav/fo/veilarbregistrering/bruker/krr/KrrClientTest.java
package no.nav.fo.veilarbregistrering.bruker.krr;

import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer;
import org.junit.jupiter.api.Test;

import java.util.Optional;

import static no.nav.fo.veilarbregistrering.FileToJson.toJson;
import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class KrrClientTest {

    private static final String OK_JSON = "/krr/hentKontaktinformasjonOk.json";
    private static final String FEIL_JSON = "/krr/hentKontaktinformasjonError.json";

    @Test
    public void skal_mappe_kontaktinfo_med_mobiltelefonnummer() {
        String json = toJson(OK_JSON);
        Foedselsnummer foedselsnummer = Foedselsnummer.of("23067844532");

        Optional<KrrKontaktinfoDto> kontaktinfoDto = KrrClient.parse(json, foedselsnummer);

        assertThat(kontaktinfoDto).isNotEmpty();
        assertThat(kontaktinfoDto.get().getMobiltelefonnummer()).isEqualTo("11111111");
    }

    @Test
    public void skal_mappe_feil_til_runtimeException() {
        String json = toJson(FEIL_JSON);
        Foedselsnummer foedselsnummer = Foedselsnummer.of("23067844539");

        RuntimeException runtimeException = assertThrows(RuntimeException.class, () -> KrrClient.parse(json, foedselsnummer));
        assertThat(runtimeException.getMessage())
                .isEqualTo("Henting av kontaktinfo fra KRR feilet: fant ikke person");
    }
}

File: src/test/java/no/nav/fo/veilarbregistrering/bruker/krr/KrrClientTest.kt
package no.nav.fo.veilarbregistrering.bruker.krr

import no.nav.fo.veilarbregistrering.FileToJson.toJson
import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.lang.RuntimeException
import kotlin.test.assertNotNull

class KrrClientTest {

    @Test
    fun `skal mappe kontaktinfo med mobiltelefonnummer med ny jsonparser`() {
        val json = toJson(OK_JSON)
        val kontaktinfoDto = KrrClient.parse(json)
        assertNotNull(kontaktinfoDto)
        assertThat(kontaktinfoDto.mobiltelefonnummer).isEqualTo("11111111")
    }

    @Test
    fun `skal mappe feil til runtimeException`() {
        val json = toJson(FEIL_JSON)
        val runtimeException = assertThrows(
            RuntimeException::class.java
        ) { KrrClient.parse(json) }
        assertThat(runtimeException.message)
            .isEqualTo("Henting av kontaktinfo fra KRR feilet: fant ikke person")
    }

    companion object {
        private const val OK_JSON = "/krr/hentKontaktinformasjonOk.json"
        private const val FEIL_JSON = "/krr/hentKontaktinformasjonError.json"
    }
}
